src/views/virtualmachines/actions/hooks/useVirtualMachineActionsProvider.ts

```diff
@@ -39,18 +45,24 @@ const useVirtualMachineActionsProvider: UseVirtualMachineActionsProvider = (vm)
       printableStatus === Paused
         ? VirtualMachineActionFactory.unpause(vm, t)
         : VirtualMachineActionFactory.pause(vm, t);
+
     return [
       startOrStop,
       VirtualMachineActionFactory.restart(vm, t),
       pauseOrUnpause,
       VirtualMachineActionFactory.clone(vm, createModal, t),
       migrateOrCancelMigration,
       // VirtualMachineActionFactory.openConsole(vm),
+      VirtualMachineActionFactory.copySSHCommand(
+        command,
+        printableStatus !== Running || !sshService,
```

Comment:
I think that if the VM won't be running, we get the username undefined. And the info about when this command should be disabled comes from also from [here](https://github.com/kubevirt-ui/kubevirt-plugin/pull/822/files#diff-e54b1f51cdd19f1a167fa1fde6f8f332e56ab7de73ceb9dae45b23c1b6c67915R45). This was already implemented by someone else before I started working on this PR.

Regarding the icon position, I am not sure. I've seen some related design doc and the icon was placed after the text. Also in the VM overview tab -> Details card -> kebab menu - there is the icon placed after the text, too. But there's a different code, different components used to display the dropdown items...

Reply:
> If the VM won't be running, we get the username undefined, for example. And I got the info about when this action should be disabled also from [here](https://github.com/kubevirt-ui/kubevirt-plugin/pull/822/files#diff-e54b1f51cdd19f1a167fa1fde6f8f332e56ab7de73ceb9dae45b23c1b6c67915R45). This was already implemented by someone else before I started working on this PR. And it makes sense to me: VM running and ssh service set. How could you get into the VM that is not running? :)

I understand what you are saying, the question is do we trust the user to not try and ssh to a stopped machine and should still allow copying the ssh command, or do we want to be nice and block it for him? and maybe add a description text when disabled to state that 'can not copy the SSH command when VirtualMachine is stopped` or something similar, this is something that is not really clear to me and it could be nice to have a clarification. also if disabling so we need to fix the wrong behavior in the VM details page :)  

> Regarding the icon position, I am not sure. I've seen the related design doc and the icon was placed after the text. Also in the VM overview tab -> Details card -> kebab menu - there is the icon already placed after the text, too. But there's a different code, different components used to display the [dropdown items](https://github.com/kubevirt-ui/kubevirt-plugin/blob/main/src/views/virtualmachines/details/tabs/overview/components/VirtualMachinesOverviewTabDetails/components/VirtualMachinesOverviewTabDetailsTitle.tsx#L42L51), different than used for the Actions, and I think this is why it was possible to place the icon after the text there. Maybe we will have to rewrite our Actions? I hope not.. Moving the icon in VM overview tab -> Details card -> kebab menu before the text would be much much easier and faster, to keep the consistency for all the menus we have.

when looking at the icon you mentioned it seems like we can add replace this [line](https://github.com/kubevirt-ui/kubevirt-plugin/blob/main/src/views/virtualmachines/list/components/VirtualMachineActions/VirtualMachineActions.tsx#L65) with this:
```
{action?.label}
    {action?.icon && <span className="text-muted">
      {' '}{action.icon}
    </span>}
```

Reply:
I think we should not allow copy the command in cases when it does not make sense. Also I would not assume that the user will be the clever one. It's always better to assume the worst case we can imagine, and to make preventive action.

Adding some description about why it is disabled makes sense to me. I just haven't found any good description yet. There's not much space so it should be something very short (if ever). Anyway, what do you mean by "wrong behavior in the VM details page"? Does disabling that option cause some bugs in the details page? I am not aware of any.

Thank you very much for the suggestion for the icon issue. Let me try it! 😊

Reply:
> what do you mean by "wrong behavior in the VM details page"? Does disabling that option cause some bugs in the details page?
I mean that if you would go in a VM that is stopped, and expose the ssh service, you would be able to copy the ssh command, if we're not allowing to copy in the vm actions or the vm overview details card kebab actions, than it shouldn't be allowed in the vm details tab under ssh access

Reply:
Well, that is a good question: what should be available to do if we stop such a VM. I think this is more designer's question.  And IMO the _SSH access_ field in the _Details_ tab is dedicated for a slightly different thing - for exposing SSH access, some setup, not just for getting the ssh command. So it feels to me okay that it is possible for a stopped VM in the Details page. But still as I said: it depends on how this feature was designed, on what customers want etc. They can get the command if they want but they won't ssh anywhere anyway, if the VM will be stopped.

Reply:
For the ssh command in the details page, we take for granted that the user knows what he's doing. If the VM is stopped he knows that would not be able to access it through ssh.
But probably he needs the command for some automations and stuff like that

Reply:
We can also change the `useSSHCommand` to take the `vm` as input instead of the `vmi` so we'll have the command also when the VM is stopped. 
This would align the logic with the `useSSHService` and would make much more sense for me

Reply:
So I've changed the `useSSHCommand` to take the `vm`. Doue to this change, I had to make some changes also in other related places.